validator/tests/test_checks.py

```diff
@@ -230,6 +230,87 @@ def test_invalid_geo_id_national(self):
         assert "US" not in validator.raised_errors[0].expression
         assert "SP" not in validator.raised_errors[0].expression
 
+class TestCheckBadGeoIdValue:
+    params = {"data_source": "", "span_length": 0,
+              "end_date": "2020-09-02", "expected_lag": {}}
+
+    def test_empty_df(self):
+        validator = Validator(self.params)
+        empty_df = pd.DataFrame(columns=["geo_id"], dtype=str)
+        validator.check_bad_geo_id_value(empty_df, "name", "county")
+        assert len(validator.raised_errors) == 0
+
+    def test_invalid_geo_id_county(self):
+        validator = Validator(self.params)
+        df = pd.DataFrame(["01001", "88888", "99999"], columns=["geo_id"])
+        validator.check_bad_geo_id_value(df, "name", "county")
+
+        assert len(validator.raised_errors) == 1
+        assert "check_bad_geo_id_value" in validator.raised_errors[0].check_data_id
+        assert len(validator.raised_errors[0].expression) == 2
+        assert "01001" not in validator.raised_errors[0].expression
+        assert "88888" in validator.raised_errors[0].expression
+        assert "99999" in validator.raised_errors[0].expression
+
+    def test_invalid_geo_id_msa(self):
+        validator = Validator(self.params)
+        df = pd.DataFrame(["10180", "88888", "99999"], columns=["geo_id"])
+        validator.check_bad_geo_id_value(df, "name", "msa")
+
+        assert len(validator.raised_errors) == 1
+        assert "check_bad_geo_id_value" in validator.raised_errors[0].check_data_id
+        assert len(validator.raised_errors[0].expression) == 2
+        assert "10180" not in validator.raised_errors[0].expression
+        assert "88888" in validator.raised_errors[0].expression
+        assert "99999" in validator.raised_errors[0].expression
+
+    def test_invalid_geo_id_hrr(self):
+        validator = Validator(self.params)
+        df = pd.DataFrame(["1", "11", "111", "8", "88", "888"], columns=["geo_id"])
+        validator.check_bad_geo_id_value(df, "name", "hrr")
+
+        assert len(validator.raised_errors) == 1
+        assert "check_geo_id_value" in validator.raised_errors[0].check_data_id
```

Comment:
```suggestion
        assert "check_bad_geo_id_value" in validator.raised_errors[0].check_data_id
```

Update due to check string/method name change.